### DISCUSSION TITLE
custom proxy contract verification on avalanche testnet fuji. avalanche c-chain

### DISCUSSION THREAD

**Discussion by waqasideofuzion** [ConsenSys-archive/truffle#5744] [General Q&A]
how can i verify the below proxy contract?

`// SPDX-License-Identifier: GPL-3.0

pragma solidity ^0.8.0;

import './authority/ITREXImplementationAuthority.sol';

contract TokenProxy {
    address public implementationAuthority;

    constructor(
        address _implementationAuthority,
        address _identityRegistry,
        address _compliance,
        string memory _name,
        string memory _symbol,
        uint8 _decimals,
        address _onchainID
    ) {
        implementationAuthority = _implementationAuthority;

        address logic = (ITREXImplementationAuthority(implementationAuthority)).getTokenImplementation();

        // solhint-disable-next-line avoid-low-level-calls
        (bool success, ) =
            logic.delegatecall(
                abi.encodeWithSignature(
                    'init(address,address,string,string,uint8,address)',
                    _identityRegistry,
                    _compliance,
                    _name,
                    _symbol,
                    _decimals,
                    _onchainID
                )
            );
        require(success, 'Initialization failed.');
    }

    fallback() external payable {
        address logic = (ITREXImplementationAuthority(implementationAuthority)).getTokenImplementation();

        assembly {
            // solium-disable-line
            calldatacopy(0x0, 0x0, calldatasize())
            let success := delegatecall(sub(gas(), 10000), logic, 0x0, calldatasize(), 0, 0)
            let retSz := returndatasize()
            returndatacopy(0, 0, retSz)
            switch success
                case 0 {
                    revert(0, retSz)
                }
                default {
                    return(0, retSz)
                }
        }
    }
}
`

**Comment by apdan510** [comment]
Hey @waqasideofuzion are you looking to run a security audit on this contract?
